src/rpc/rpc_reward_api.py

```diff
@@ -138,32 +139,28 @@ def __get_delegators_and_delgators_balance(self, cycle, current_level):
         delegate_staking_balance = 0
         delegators = {}
 
+        # try to fetch delegate's staking balance and list of delegators
         try:
             response = self.do_rpc_request(request)
             delegate_staking_balance = int(response["staking_balance"])
-
             delegators_addresses = response["delegated_contracts"]
-            for idx, delegator in enumerate(delegators_addresses):
-                request = self.COMM_DELEGATE_BALANCE.format(self.node_url, hash_snapshot_block, delegator)
-                response = self.do_rpc_request(request)
-
-                sleep(0.5)  # be nice to public node service
 
-                response = None
+        except:
+            logger.warn('No delegators or unexpected error while fetching delegators', exc_info=True)
 
-                while not response:
-                    try:
-                        response = self.do_rpc_request(request, time_out=5)
-                    except:
-                        logger.debug("Fetching delegator info failed {}, will retry", delegator)
+        # loop over found delegators
+        for idx, delegator in enumerate(delegators_addresses):
+            request = self.COMM_DELEGATE_BALANCE.format(self.node_url, hash_snapshot_block, delegator)
 
-                delegators[delegator] = int(response["balance"])
+            try:
```

Comment:
Shouldn't the loop around this fetch operation be kept? (in case a large number of delegators, one or more rpc call(s) may fail, if this fetch is not retried, the balance will be considered as zero (https://github.com/habanoz/tezos-reward-distributor/commit/4494abf7e2f874d18f57757560d21f10dbb953bc)

Reply:
Yes, there should be some retry logic. But what was there previously made RPC calls twice for no reason. If you look here, https://github.com/habanoz/tezos-reward-distributor/blob/d1cfad3d1c7738a999c381ca25afd74cd04aaf8d/src/rpc/rpc_reward_api.py#L148, you see that request is set, then it sleeps, then sets to none, then does it again. That's a complete waste of a request.

I'll add the loop back in but without calling RPC twice